### ISSUE TITLE
Empty data set view breaks "pull to refresh"

### ISSUE THREAD

**Issue by leoschweizer** [dzenbot/DZNEmptyDataSet#43]
I'm trying to use this handy library in a `UITableViewController` which applies the "pull to refresh" pattern and which thus owns a `UIRefreshControl`. Unfortunately, this combination causes a number of problems:

1) As long as the empty data set view is visible, the "pull to refresh" gesture is no longer possible even when `emptyDataSetShouldAllowTouch:` returns `NO`.

2) The empty data set view seems to hide the `UIRefreshControl` (which can be invoked manually by calling `beginRefreshing`, but it won't become visible as long as the empty data set view is visible). This is also the case when I manually set the background color to transparent (but as far as I can judge from looking at the source code this is the default anyway).

3) The offset of the empty data set view is wrong (although this could probably also be fixed by returning a custom offset from the according delegate method).

Is there any way to use this library in conjunction with a `UIRefreshControl`? If there is anything I can do to help resolve this issue, just let me know.

**Comment by allaire** [comment]
@leoschweizer Did you fix it?

**Comment by leoschweizer** [comment]
@allaire No, I figured that simply using the background view of a `UITableView` is a much more reliable solution for my use case than working around the design of this library.

**Comment by rene-dohan** [comment]
Tried UIRefreshControl with 
`- (BOOL)emptyDataSetShouldAllowScroll:(UIScrollView *)scrollView {

    return YES;
}`

but it just don't work...

**Comment by Artheyn** [comment]
@allaire @rene-dohan Working with a CollectionView I had to implement these two delegates in order to restore the pull-to-refresh gesture (even if the `emptyDataSetShouldAllowTouch` is marked as `true` by default... Weird but it works.):

```swift
    func emptyDataSetShouldAllowScroll(_ scrollView: UIScrollView!) -> Bool {
        return true
    }

    func emptyDataSetShouldAllowTouch(_ scrollView: UIScrollView!) -> Bool {
        return true
    }
```

Hope it helps.

**Comment by ushernut** [comment]
In case anyone made mistake like me, with the override method above, please make sure you have set self.tableview.emptyDataSetDelegate = self
Without this line, refresh control will never show up.

**Comment by kolyan94** [comment]
Maybe it'll help someone as a temp solution.
` 
func emptyDataSetShouldAllowScroll(_ scrollView: UIScrollView!) -> Bool {
    return true
  }
  
  func scrollViewDidScroll(_ scrollView: UIScrollView) {
    guard let view = scrollView.value(forKey: "emptyDataSetView") as? UIView else {return}
    view.frame = CGRect(x: view.frame.origin.x, y: scrollView.contentOffset.y, width: view.bounds.width, height: view.bounds.height)
  }
`
